parkdatetimepicker/src/main/java/com/smparkworld/parkdatetimepicker/core/DateTimeModeNavigatorImpl.kt

```diff
@@ -0,0 +1,116 @@
+package com.smparkworld.parkdatetimepicker.core
+
+import com.smparkworld.parkdatetimepicker.extension.addRequiredNonNullItem
+import com.smparkworld.parkdatetimepicker.model.BaseListener
+import com.smparkworld.parkdatetimepicker.model.PhaseTransactionData
+import com.smparkworld.parkdatetimepicker.model.SelectedDate
+import com.smparkworld.parkdatetimepicker.model.SelectedTime
+import com.smparkworld.parkdatetimepicker.ui.bottomsheet.datetime.model.DateTimeMode
+import com.smparkworld.parkdatetimepicker.ui.bottomsheet.datetime.model.Phase
+
+internal class DateTimeModeNavigatorImpl(
+    private val listenerManager: ListenerManager = ListenerManagerImpl()
+): DateTimeModeNavigator {
+
+    private lateinit var mode: DateTimeMode
+
+    private var currentPhase: Phase = Phase.INIT
+
+    private val selectedDates = mutableListOf<SelectedDate>()
+    private val selectedTimes = mutableListOf<SelectedTime>()
+
+    override fun init(mode: DateTimeMode, listener: BaseListener?): PhaseTransactionData {
+        this.mode = mode
+        listenerManager.init(mode, listener)
+
+        return getNextPhase()
+    }
+
+    override fun getNextPhase(
+        selectedDate: SelectedDate?,
+        selectedTime: SelectedTime?
+    ): PhaseTransactionData {
+        assertInitialized()
+        return when(mode) {
+            DateTimeMode.DATETIME -> {
+                when(currentPhase) {
+                    Phase.INIT -> Phase.DATE
+                    Phase.DATE -> {
+                        selectedDates.addRequiredNonNullItem(selectedDate)
+                        Phase.TIME
+                    }
+                    Phase.TIME -> {
+                        selectedTimes.addRequiredNonNullItem(selectedTime)
+                        Phase.DONE
+                    }
```

Comment:
Consider changing the phase class to a sealed class later. And consider adding a value such as SelectedDate to the sealed class.

It looks a bit risky code.. 8ㅅ8